code/__HELPERS/turfs.dm

```diff
@@ -279,6 +279,67 @@ Turf and target are separate in case you want to teleport some distance from a t
 	LAZYSET(modifiers, ICON_Y, "[(click_turf_py - click_turf.pixel_y) + ((click_turf_y - click_turf.y) * ICON_SIZE_Y)]")
 	return click_turf
 
+/**
+ * Takes mousepos_x and mousepos_y and the client that provided them and returns the turf clicked on
+ * reminder, mouseposx and mouseposy start from the top left corner because we can't have consistency or nice things
+ * they also ignore sizes, so we need to compensate for that
+ * sizex and sizey are the size of the map for this purpose
+ * also keep in mind that the screen size is the worlds pixel size, while mousepos and size are the completely unrelated control size
+ */
+/proc/get_loc_from_mousepos(mousepos_x, mousepos_y, sizex, sizey, client/viewing_client)
+	//hack because we can't have shit with this engine!!!!!!
+	// https://discord.com/channels/725444629172060262/1063155296052191373/1357460149836451960
```

Comment:
Would prefer if you elaborated fully instead of linking a discord post that may get lost to time/deleted/whatever in time